state/controller.go

```diff
@@ -246,23 +246,23 @@ func readRawControllerInfo(session *mgo.Session) (*ControllerInfo, error) {
 const stateServingInfoKey = "stateServingInfo"
 
 // StateServingInfo returns information for running a controller machine
-func (st *State) StateServingInfo() (StateServingInfo, error) {
+func (st *State) StateServingInfo() (jujucontroller.StateServingInfo, error) {
 	controllers, closer := st.db().GetCollection(controllersC)
 	defer closer()
 
-	var info StateServingInfo
+	var info jujucontroller.StateServingInfo
```

Comment:
I think we should have a different struct defined in the state package that has the bson serialization tags.

Use that to get things in and out of the database, but return the controller.StateServingInfo as we are now.